util.py

```diff
@@ -419,6 +421,93 @@ def winedll_override(dll: str, dtype: Literal['n', 'b', 'n,b', 'b,n', '']) -> No
     )
 
 
+def patch_libcuda() -> bool:
+    """Patches libcuda to work around games that crash with DLSS due to wonky memory allocation.
```

Comment:
I'm fine having this function because I think it would be useful. Hopefully this fix is short lived and it gets properly addressed by the CUDA engineers or some other project upstream.

I'm interested in more context and the scale of this problem though. For instance, does this occur in Windows as well and does this effect every game when DLSS is enabled?

Reply:
I'm not too well versed in CUDA, let alone the nvidia driver stack, but as far as I can tell from a distance this seems to be a difference in the approach to memory allocation in Linux versus Windows. 

> When creating a Vulkan device with VK_NVX_binary_import and/or VK_NVX_image_view_handle extensions... libcuda tries to allocate memory in a specific area already used by the game, so... failing.
DXVK and VKD3D get around this by recreating the device without these extensions... but without these extensions DLSS does not work.
See: https://github.com/jp7677/dxvk-nvapi/issues/174#issuecomment-2227462795 onwards for additional context (seems like a conversation between an NVIDIA employee/contractor and the individual who found a workaround to the issue.)

This could be rearing it's head due to a Linux specific platform issue (with those two above extensions) or it could also be that the CUDA stack on Windows protects itself better against issues like this, probably due to the size/ demographics of the userbase.

Anyhow hope that gives you some context as to why this is needed for Star Citizen at the moment (and potentially others if this implementation issue is repeated in the future)